rackcity/permissions/permissions.py

```diff
@@ -85,6 +85,9 @@ def validate_user_asset_permission_to_add(user, validated_data):
     # TODO: make this check work for blades, which don't have a rack
     if "rack" not in validated_data:
         return
+    ## ading this to throw readable error if no rack specified
+    if not validated_data["rack"]:
+        raise Exception("Rack must be specified")
```

Comment:
adding a readable error message for when no rack is selected on asset creation.

Reply:
this doesn't really have to do with permissions, since it just means the request being sent is invalid, so I think it should go in asset_add method and not here

Reply:
removing this and sending this to ben so he can also handle similar logic if no chassis is specified